pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -97,9 +97,14 @@ func (admitter *VMsAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1
 		return webhookutils.ToAdmissionResponseError(err)
 	}
 
-	causes := admitter.applyFlavorToVm(&vm)
-	if len(causes) > 0 {
-		return webhookutils.ToAdmissionResponse(causes)
+	var causes []metav1.StatusCause
+	// validate flavor only during creation time
+	if vm.UID == "" {
+		causes = admitter.applyFlavorToVm(&vm)
+		if len(causes) > 0 {
+			return webhookutils.ToAdmissionResponse(causes)
```

Comment:
If I remember correctly, mutating webhooks can also fail admission, so you could move this check to the mutating webhook.

Reply:
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go#L38
I am not sure if this comment is still relevant

Reply:
The upstream documentation states that minimal required version is 1.10:
https://kubevirt.io/user-guide/operations/installation/#requirements

I'm not sure if that is still correct.

Reply:
This validating webhook will not work, because the flavor was already applied to the VM in the mutating webhook, so here `admitter.applyFlavorToVm()` will always return conflicts.

Reply:
> This validating webhook will not work, because the flavor was already applied to the VM in the mutating webhook, so here admitter.applyFlavorToVm() will always return conflicts.

yup, so after expanding the flavor into the VM during mutation webhook, the only check we need to make in the validation webhook is ensuring that the flavor matching is never mutated to another flavor, We'd only allow removal of the flavor after VM creation.